src/libs/auth.js

```diff
@@ -183,26 +184,55 @@ export const refreshTerraProfile = async () => {
   authStore.update(state => ({ ...state, profile }))
 }
 
+//helpers for external services statuses
+
+const loadStatus = async func => {
+  try {
+    return await func
+  } catch (error) {
+    if (error.status === 404) {
+      return {}
+    } else {
+      throw error
+    }
+  }
+}
+
+const fenceNotify = (status, oldStatus, notificationId, serviceName, provider) => {
+  if (status !== oldStatus) {
+    const redirectUrl = `${window.location.origin}/${Nav.getLink('fence-callback')}`
+
+    const now = Date.now()
+    const expireTime = status && addDays(30, parseJSON(status.issued_at))
+    const fiveDays = 1000 * 60 * 60 * 24 * 5
+    const expireStatus = Utils.cond(
+      [!expireTime, () => null],
+      [now >= expireTime, () => 'has expired'],
+      [now > expireTime - (fiveDays), () => `will expire in ${getDay(expireTime - (fiveDays))} days`]
```

Comment:
The calculation of the message here is not showing what we want. `getDay` shows the day of the week, where we probably want to use `differenceInDays` between `now` and `expireTime`.